test/data/basic-job-with-template.yaml

```diff
@@ -0,0 +1,7 @@
+jobs:
+    main:
+        template: mytemplate@1.2.3
```

Comment:
Let's not handle overriding yet. For overriding, we need to go through each step in individual job and look up the exact step inside template, then override that specific step while keeping everything else the same. Since this logic is more complicated than replacing the whole step object, we will do it in 2nd pass when we handle override & wrapping of steps.